tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/GraphQL/HotChocolate/ExecuteAsyncIntegration.cs

```diff
@@ -0,0 +1,82 @@
+// <copyright file="ExecuteAsyncIntegration.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.ComponentModel;
+using System.Linq;
+using System.Threading;
+using Datadog.Trace.ClrProfiler.CallTarget;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.GraphQL.HotChocolate
+{
+    /// <summary>
+    /// GraphQL.Execution.ExecutionStrategy calltarget instrumentation
+    /// </summary>
+    [InstrumentMethodAttribute(
+        IntegrationName = HotChocolateCommon.IntegrationName,
+        MethodName = HotChocolateCommon.ExecuteAsyncMethodName,
+        ReturnTypeName = "System.Threading.Tasks.Task`1<HotChocolate.Execution.IExecutionResult>",
+        ParameterTypeNames = new string[] { "HotChocolate.Execution.IQueryRequest", "System.Threading.CancellationToken" },
+        AssemblyName = HotChocolateCommon.HotChocolateAssembly,
+        TypeName = "HotChocolate.Execution.RequestExecutor",
+        MinimumVersion = HotChocolateCommon.Major12,
+        MaximumVersion = "12.*.*")]
+    [Browsable(false)]
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class ExecuteAsyncIntegration
+    {
+        /// <summary>
+        /// OnMethodBegin callback
+        /// </summary>
+        /// <typeparam name="TTarget">Type of the target</typeparam>
+        /// <typeparam name="TQueyRequest">Type of the queryRequest</typeparam>
+        /// <param name="instance">Instance value, aka `this` of the instrumented method.</param>
+        /// <param name="request">QueryRequest</param>
+        /// <param name="token">Cancellation token</param>
+        /// <returns>Calltarget state value</returns>
+        internal static CallTargetState OnMethodBegin<TTarget, TQueyRequest>(TTarget instance, TQueyRequest request, CancellationToken token)
+            where TQueyRequest : IQueryRequest
+        {
+            return new CallTargetState(scope: HotChocolateCommon.CreateScopeFromExecuteAsync(Tracer.Instance, request), state: instance);
```

Comment:
You can avoid passing the instance as state, instance is available in the end method